proof/crefine/ARM_HYP/Wellformed_C.thy

```diff
@@ -43,7 +43,7 @@ abbreviation
 abbreviation
   pd_Ptr :: "32 word \<Rightarrow> (pde_C[2048]) ptr" where "pd_Ptr == Ptr"
 abbreviation
-  regs_C_Ptr :: "addr \<Rightarrow> (machine_word_len word[42]) ptr" where "regs_C_Ptr \<equiv> Ptr"
+  regs_C_Ptr :: "addr \<Rightarrow> (machine_word_len word[43]) ptr" where"regs_C_Ptr \<equiv> Ptr"
```

Comment:
I think I did something to improve this on AARCH64. Might be worth seeing what the lines you changed in this PR look like there in case there's improvement that we already figured out.

Reply:
ah yes, the `value_type vcpuregs_count = "length enum_vcpureg"` failing; so I tried to but did not succeed. any chance you could have a stab at that now?

Reply:
Hm, it probably needs some code generator setup and be done outside the Arch locale. I can have a look.

Reply:
It turns out there is a constant in C `seL4_VCPUReg_Num` for which we can declare `seL4_VCPUReg_Num_def[code]` and that then works fine.